carlosesh/Todo
# Update To Use custom Axe reporter

### Title: Enhancing Accessibility Reporting with Browser Information

### Description:

This Pull Request introduces significant improvements to our accessibility reporting by including detailed information about the tests and the browsers used. The changes are aimed at providing more context to the accessibility violations and making the reports more informative.

#### Key Changes:

1. **Custom Reporter Class**: Introduced a custom reporter class `AxeHtmlReporter` that integrates with Playwright and axe-core to generate HTML reports for accessibility violations.

2. **Browser Information**: Included the browser information (e.g., Chrome, Firefox) in the report. This helps in understanding how the accessibility checks perform across different browsers.

3. **Grouping by Test**: Grouped the accessibility violations by test, making it easier to correlate the violations with specific test cases.

4. **Custom Summary**: Utilized the custom summary feature of `axe-html-reporter` to include a summary section in the report, providing additional details about the tests and browsers used.

5. **Code Refactoring**: Refactored the code to make it more modular and maintainable, including proper handling of attachments and grouping logic.

#### How to Test:

1. Run the accessibility tests as usual.
2. Check the generated HTML report in the specified output directory.
3. Verify that the report includes the test names, browser information, and accessibility violations in a well-structured format.

#### Screenshots (if applicable):
![_C__Users_carlo_WebstormProjects_demo-app_reports_axe-report_index html (1)](https://github.com/carlosesh/Todo/assets/20198209/de1e3df5-8938-4846-ba99-444218799d9b)
